pennylane/data/dataset.py

```diff
@@ -0,0 +1,185 @@
+
+        >>> dataset.write('./path/to/file/dataset.dat')
+
+        We can then retrieve the data using :func:`Dataset.read()`
+
+        >>> retrieved_data = qml.data.Dataset()
+        >>> retrieved_data.read('./path/to/file/dataset.dat')
+        >>> print(retrieved_data.Hamiltonian)
+          (1) [Z0]
+        + (1) [Z1]
+        >>> print(dataset.Hamiltonian)
+          (1) [Z0]
+        + (1) [Z1]
+    """
+
+    def __init__(self, dtype=None, **kwargs):
+        self.dtype = dtype
+        self.__doc__ = ""
+        for key, val in kwargs.items():
+            setattr(self, f"{key}", val)
+
+    def __eq__(self, __o):
+        return self.__dict__ == __o.__dict__
+
+    @staticmethod
+    def _read_file(filepath):
+        """Loads data previously saved with :func:`~pennylane.data.dataset.write`.
+        Data is read as a dictionary."""
+        with open(filepath, "rb") as file:
+            compressed_pickle = file.read()
+
+        depressed_pickle = zstd.decompress(compressed_pickle)
+        data = dill.loads(depressed_pickle)
+        return data
+
+    def read(self, filepath):
+        """Loads data from a saved file to the current dataset.
+
+        Args:
+            filepath (string): the desired location and filename to load, e.g. './path/to/file/file_name.dat'.
+
+        **Example**
+
+        We first initialize a :class:`~pennylane.data.Dataset` and then read saved data to it:
+
+        .. code-block:: python
+
+            import pennylane as qml
+
+            new_dataset = qml.data.Dataset(kw1 = 1, kw2 = '2', kw3 = [3])
+            new_dataset.read('./path/to/file/file_name.dat')
+        """
+        data = self._read_file(filepath)
+        for (key, val) in data.items():
+            setattr(self, f"{key}", val)
+
+    @staticmethod
+    def _write_file(filepath, data, protocol=4):
+        """Writes the input data to a file as a dictionary."""
+        pickled_data = dill.dumps(data, protocol=protocol)  # returns data as a bytes object
+        compressed_pickle = zstd.compress(pickled_data)
+        with open(filepath, "wb") as file:
+            file.write(compressed_pickle)
+
+    def write(self, filepath, protocol=4):
+        """Writes a dataset to a file as a dictionary.
+
+        Args:
+            filepath (string): the desired save location and file name, e.g. :code:'./path/to/file/file_name.dat'.
+
+        **Example**
+
+        We can save an existing dataset to a specific file as follows:
+
+        .. code-block:: python
+
+            import pennylane as qml
+
+            new_dataset = qml.data.Dataset(kw1 = 1, kw2 = '2', kw3 = [3])
+            new_dataset.write('./path/to/file/file_name.dat')
+        """
+        dataset = {}
+        for (key, val) in self.__dict__.items():
+            dataset.update({key: val})
+        self._write_file(filepath=filepath, data=dataset, protocol=protocol)
+
+    @classmethod
+    def from_dataset(cls, dataset):
```

Comment:
How about a name such as `copy`:

`qml.data.Dataset.copy(original_dataset)`

In general, we can use simple function names such as `read`, `write`, `load`, `copy`, etc.

Reply:
You guys are right! I will modify this.

Reply:
Is it possible to have a syntax like the following, or it will break things:

```
qml.data.read()
qml.data.write()
...
```